op-signer.md

```diff
@@ -0,0 +1,280 @@
+
+```
+key-ring-name: op-signer
+Multi-region: Global
+```
+
+Then click `CREATE`.
+
+### Create a Key
+
+Choose a key ring on the page https://console.cloud.google.com/security/kms and click `CREATE KEY`. In the `Create Key` page, assign values as in the following example and click `CONTINUE` for each step:
+
+```
+key name: op-challenger (Depending on the consumer of the key)
+Protection Level: HSM (Protection level must be set to HSM to use secp256k1)
+Key material: HSM-generated key
+Purpose: Asymmetric sign
+Algorithm: Elliptic Curve secp256k1 - SHA256 Digest (Ethereum compatible)
+Default expiration: 30 days
+```
+
+Then click `CREATE`.
+
+### Grant Permissions
+
+Click the `Permissions` tab of the Key detail page and click `Grant Access`.
+
+- Add the email address of the service account as principals from [this step](#create-a-service-account) and assign proper roles. For example:
+
+```
+Add principals: swc-signer-sa@signing-test-450710.iam.gserviceaccount.com
+Assign roles: roles/cloudkms.signerVerifier
+```
+
+Note that only IAM principals with Owner (roles/owner) or Cloud KMS Admin (roles/cloudkms.admin) roles can grant or revoke access to Cloud KMS resources.
+
+### Get Resource Name of the Key
+
+Click on the created item and in the `Versions` tab of the key detail page, click the button under `Actions` and select `Copy resource name`.
+
+For example:
+
+```
+projects/swc-signer/locations/global/keyRings/op-signer/cryptoKeys/op-challenger/cryptoKeyVersions/1
+```
+
+You will add this value as `key` in `op-signer/config.yaml` in the `infra` repo.
+
+## Deploy op-signer Service
+
+### Build the Source
+
+Log on to the server where the op-signer service will be deployed.
+
+```bash
+git clone https://github.com/QuarkChain/infra.git
+cd infra/op-signer
+make
+```
+
+### Generate Server TLS
+
+While in `infra/op-signer`, set the domain name or IP address of the server as the value of `SIGNER_SERVER_HOST` environment variable, and generate TLS:
+
+```bash
+export SIGNER_SERVER_HOST=65.108.236.27
+./tls-server.sh
+```
+
+You will receive several CA and TLS-related files in `tls-server` folder:
+
+```bash
+tls-server
+├── ca.crt
+├── ca.key
+├── ca.srl
+├── tls.crt
+├── tls.csr
+└── tls.key
+```
+> ⚠️ **Note:** 
+> If the CA key is compromised, an attacker can issue any certificate using it, which will be recognized as legitimate by the system.
+
+### Configure Google API Credentials
+
+Add the following line to `op-signer/.envrc` in the `infra` repo:
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS="<PATH_TO_SERVICE_ACCOUNT_JSON_FILE>"
+```
+
+Replace `<PATH_TO_SERVICE_ACCOUNT_JSON_FILE>` with the path of the service account JSON file downloaded in [this step](#get-service-account-key-file).
```

Comment:
If the hacker can get the access of SERVICE_ACCOUNT_JSON_FILE, he can request any signing data